test/jasmine/tests/modebar_test.js

```diff
@@ -339,25 +340,32 @@ describe('ModeBar', function() {
         it('updates mode bar buttons if modeBarButtonsToRemove changes', function() {
             var gd = setupGraphInfo();
             manageModeBar(gd);
+            var initialButtonCount = countButtons(gd._fullLayout._modeBar);
 
             gd._context.modeBarButtonsToRemove = ['toImage', 'sendDataToCloud'];
             manageModeBar(gd);
 
-            expect(countButtons(gd._fullLayout._modeBar)).toEqual(9);
+            expect(countButtons(gd._fullLayout._modeBar))
+                .toEqual(initialButtonCount - 2);
         });
 
         it('updates mode bar buttons if modeBarButtonsToAdd changes', function() {
             var gd = setupGraphInfo();
             manageModeBar(gd);
 
+            var initialGroupCount = countGroups(gd._fullLayout._modeBar),
+                initialButtonCount = countButtons(gd._fullLayout._modeBar);
+
             gd._context.modeBarButtonsToAdd = [{
                 name: 'some button',
                 click: noop
             }];
             manageModeBar(gd);
 
-            expect(countGroups(gd._fullLayout._modeBar)).toEqual(6);
-            expect(countButtons(gd._fullLayout._modeBar)).toEqual(12);
+            expect(countGroups(gd._fullLayout._modeBar))
+                .toEqual(initialGroupCount + 1);
+            expect(countButtons(gd._fullLayout._modeBar))
+                .toEqual(initialButtonCount + 1);
```

Comment:
I started trying to make these tests more robust against changes... but was only convenient to do for this change test.